rules/no-useless-promise-resolve-reject.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+const {methodCallSelector} = require('./selectors/index.js');
+
+const RETURN_RESOLVE = 'return-resolve';
+const RETURN_REJECT = 'return-reject';
+const YIELD_RESOLVE = 'yield-resolve';
+const YIELD_REJECT = 'yield-reject';
+const messages = {
+	[RETURN_RESOLVE]: 'Prefer `return value` over `return Promise.resolve(value)`.',
+	[RETURN_REJECT]: 'Prefer `throw error` over `return Promise.reject(error)`.',
+	[YIELD_RESOLVE]: 'Prefer `yield value` over `yield Promise.resolve(value)`.',
+	[YIELD_REJECT]: 'Prefer `throw value` over `yield Promise.reject(value)`.',
+};
+
+const promiseResolveRejectSelector = methodCallSelector({
+	object: 'Promise',
+	methods: ['resolve', 'reject'],
+	allowComputed: true,
+});
+const asyncArrowFunctionReturnSelector = `ArrowFunctionExpression[async=true] > ${promiseResolveRejectSelector}`;
+const returnStatementSelector = `ReturnStatement > ${promiseResolveRejectSelector}`;
+const yieldExpressionSelector = `YieldExpression > ${promiseResolveRejectSelector}`;
+
+const functionTypes = new Set([
+	'ArrowFunctionExpression',
+	'FunctionDeclaration',
+	'FunctionExpression',
+]);
+const getParentFunction = node => {
+	let {parent} = node;
+	while (parent && !functionTypes.has(parent.type)) {
+		parent = parent.parent;
+	}
+
+	return parent;
+};
+
+/** @param {import('eslint').Rule.RuleContext} context */
+const create = context => {
+	const sourceCode = context.getSourceCode();
+
+	/** @param {(isResolve: boolean, valueString: string, value) => import('eslint').Rule.ReportFixer} createFix */
+	const createProblem = (node, createFix, resolveMessage = RETURN_RESOLVE, rejectMessage = RETURN_REJECT) => {
+		const isResolve = node.callee.property.name === 'resolve';
+		const value = node.arguments.length === 1 ? node.arguments[0] : undefined;
+		return {
+			node,
+			messageId: isResolve ? resolveMessage : rejectMessage,
+			fix: value === undefined ? undefined : createFix(isResolve, sourceCode.getText(value), value),
```

Comment:
`sourceCode.getText(value)` should try keep parens.

```js
async () => Promise.resolve(  (0, 1) )
```

Should fix to 

```js
async () => (0, 1)
```
Not

```js
async () => 0, 1
```

Reply:
Should parentheses be preserved even if they are not required (e.g. `async () => Promise.resolve((foo))` → `async () => (foo)`?

Reply:
I guess `ObjectExpression` and `SequenceExpression` check is good enough, we can remove or keep, they are both valid.